pkg/storegateway/pool.go

```diff
@@ -54,3 +61,51 @@ func (p *chunkBytesPool) Get(sz int) (*[]byte, error) {
 func (p *chunkBytesPool) Put(b *[]byte) {
 	p.pool.Put(b)
 }
+
+// chunkSlicePool is a memory pool of chunk slices.
+type chunkSlicePool struct {
+	pool sync.Pool
+}
+
+// newChunksSlicePool creates a new chunks pool.
+func newChunksSlicePool() *chunkSlicePool {
```

Comment:
I think `newChunkSlicePool` better reflects the type returned (`chunkSlicePool`) :)
```suggestion
// newChunkSlicePool creates a new chunkSlicePool.
func newChunkSlicePool() *chunkSlicePool {
```